### PR TITLE
fix: SvgComponent getting blurred and pixelized

### LINKED ISSUE #1797: SvgComponent getting blurred and pixelized
# Current bug behaviour

SvgComponent renders blurry image.

On physical iPhone X device, image seems to be brightened and pixelized, like it was first rendered and then scaled up.
Top image is from flutter_svg, bottom image is from SvgComponent; 

blur_iphone_x

On iPhone SE simulator, the results are similar;  flutter_svg renders fine, SvgComponent blurries image.

blur_iphone_simulator

# Expected behaviour

I expected SvgComponent to render sharp, razor-edged paths, just like flutter_svg.
# Steps to reproduce

Clone the project and run on a physical device
https://github.com/realvotum/flame_svg_example

# Flutter doctor output

```
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3.1 21E258 darwin-arm, locale pl-PL)
    • Flutter version 3.0.0 at /Users/mateusz/fvm/versions/3.0.0
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (9 weeks ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/mateusz/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Connected device (3 available)
    • iPhone SE (3rd generation) (mobile) • D306D4C0-F556-4FD0-BAFC-66BB43E9A4A8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-4 (simulator)
    • macOS (desktop)                     • macos                                • darwin-arm64   • macOS 12.3.1 21E258 darwin-arm
    • Chrome (web)                        • chrome                               • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

```

# More environment information
```
* Flame version: 1.1.1
* Flame SVG version: 1.2.0
* Flutter SVG version: 1.1.0
* Platform affected: ios, not tested on others
* Platform version affected: ios 15
```

### PR DESCRIPTION
**Pull request by trunghq3101**
# Description

Since the `PictureRecorder` turns the output into pixels it will be blurry if the size isn't scaled with the `devicePixelRatio`.
This PR applies the same technique implemented in https://github.com/flame-engine/flame/pull/878 and described in https://github.com/flutter/flutter/issues/17782#issuecomment-390837913

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #1797

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,5 @@ coverage
 Makefile
 !doc/_sphinx/Makefile
 .bak
+packages/flame_svg/example/macos
+packages/flame_svg/example/windows
```
Comment (spydon) [inline .gitignore:32]: These should be in the `flame_svg` .gitignore

File: packages/flame_svg/dart_test.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+tags:
+  golden:
```
Comment (spydon) [inline packages/flame_svg/dart_test.yaml:2]: Is this file really needed? We don't have it for our other goldens

**Review by spydon — APPROVED** [review]
Good job!!

File: .gitignore (hunk 1 above)
Comment (trunghq3101) [reply .gitignore]: I moved them to `flame_svg/example` .gitignore

File: packages/flame_svg/dart_test.yaml (hunk 2 above)
Comment (trunghq3101) [reply packages/flame_svg/dart_test.yaml]: No, I removed it.

Comment (trunghq3101) [reply packages/flame_svg/dart_test.yaml]: Actually, this warning will be shown when running `flutter test` if I remove the file.
```
Warning: A tag was used that wasn't specified in dart_test.yaml.
  golden was used in the test "render sharply"
```
It is also documented in the `golden_toolkit` package ([here](https://pub.dev/packages/golden_toolkit#:~:text=Add%20a%20%22golden%22%20tag%20to%20your%20project)), so I think we should keep it.

Comment (spydon) [reply packages/flame_svg/dart_test.yaml:2]: Aah, seemingly we don't use `golden_toolkit`, but `testGolden` from `flame_test`. Maybe you could migrate to that so that we are consistent between the packages?

Comment (trunghq3101) [reply packages/flame_svg/dart_test.yaml]: I updated the test and removed `golden_toolkit`. Ended up using `flutter_test` because I need to show the Svg widget rendered by `flutter_svg` package as well.

**Review by spydon — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]
